flink-yarn/src/main/scala/org/apache/flink/yarn/YarnJobManager.scala

```diff
@@ -155,12 +156,14 @@ class YarnJobManager(
 
       rmClientOption foreach {
         rmClient =>
-          Try(rmClient.unregisterApplicationMaster(status, diag, "")).recover{
+          Try(rmClient.unregisterApplicationMaster(status, diag, "")).recover {
+            // https://issues.apache.org/jira/browse/YARN-1842
+            case e : InvalidApplicationMasterRequestException =>
```

Comment:
Okay, I think the cleaner solution is to ignore the shutdown request callback. This changes nothing about the current behavior of the YarnJobManager and keeps the code clean of such unfortunate workarounds. Do you approve?